samples/gibbs_ensemble/gibbs.py

```diff
@@ -0,0 +1,139 @@
+import socket
+import struct
+import pickle
+import numpy as np
+
+
+class MessageId:
+    START = 0
+    END = 1
+    MOVE_PART = 2
+    MOVE_PART_REVERT = 21
+    CHANGE_VOLUME = 3
+    CHANGE_VOLUME_REVERT = 31
+    EXCHANGE_PART_ADD = 4
+    EXCHANGE_PART_ADD_REVERT = 41
+    EXCHANGE_PART_REMOVE = 5
+    EXCHANGE_PART_REMOVE_REVERT = 51
+    ENERGY = 6
+
+
+class Client():
+    '''
+    Base class for a Gibbs ensemble client. Each client runs a single
+    system, executes requested changes and returns energies.
+    '''
+    _system = None
+
+    def __init__(self, system):
+        self._system = system
+
+    def random_particle(self):
+        id = np.random.choice(self._system.part[:].id)
+        return self._system.part[id]
```

Comment:
why not ```return np.random.choice(self._system.part)```

Reply:
Doc states it requires an 1d array, which ParticleList is not.

Reply:
https://docs.python.org/3/library/random.html#random.choice